packages/key-management/src/KeyAgentBase.ts

```diff
@@ -58,8 +58,21 @@ export abstract class KeyAgentBase implements KeyAgent {
   /**
    * See https://github.com/cardano-foundation/CIPs/tree/master/CIP-1852#specification
    */
-  async deriveAddress({ index, type }: AccountAddressDerivationPath): Promise<GroupedAddress> {
-    const knownAddress = this.knownAddresses.find((addr) => addr.type === type && addr.index === index);
+  async deriveAddress(
+    { index, type }: AccountAddressDerivationPath,
+    stakeKeyDerivationIndex?: number
```

Comment:
Maybe let's replace STAKE_KEY_DERIVATION_PATH with this, so that functions that have an optional stake key index argument can just set it as default value
```suggestion
    stakeKeyDerivationIndex: number = DEFAULT_STAKE_KEY_DERIVATION_INDEX
```